packages/forms/backend/sirius-components-widget-reference/src/main/java/org/eclipse/sirius/components/widget/reference/dto/CreateElementHandlerInput.java

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.widget.reference.dto;
+
+import java.util.UUID;
+
+import org.eclipse.emf.ecore.EObject;
+
+/**
+ * Input object to pass inputs to the createElementHandler field of ReferenceWidget.
+ *
+ * @author Jerome Gout
+ */
+public record CreateElementHandlerInput(UUID documentId, String domainId, EObject parent, String creationDescriptionId) {
```

Comment:
That's quite odd to have records named XxxInput which do not behave like any of our inputs (do not implement IInput and are not used as input). Could the name evolve to ensure that nobody would confuse them with our inputs (same comment for `MoveReferenceValueHandlerInput`). Input classes mean something in our application.

Reply:
Maybe CreateElementHandlerParameters.